app/jobs/send_gpo_code_expiration_notices_job.rb

```diff
@@ -0,0 +1,77 @@
+class SendGpoCodeExpirationNoticesJob < ApplicationJob
+  queue_as :low
+
+  def initialize(analytics: nil)
+    @analytics = analytics
+  end
+
+  def perform
+    codes_to_send_notifications_for.find_each do |code|
+      user = code.profile.user
+
+      user.send_email_to_all_addresses(:gpo_code_expired, code_sent_at: code.code_sent_at)
+      code.update(expiration_notice_sent_at: Time.zone.now)
+
+      analytics.idv_gpo_expiration_email_sent(user_id: user.uuid)
+    end
+  end
+
+  def codes_to_send_notifications_for
+    # We are looking at a 48 hr window in which all codes will _definitely_ be expired.
+    from, to = calculate_notification_window_bounds
+    expired_codes_needing_notification_sent_between(
+      from: from,
+      to: to,
+    )
```

Comment:
this fits under 100c, maybe we shorten it?
```suggestion
    expired_codes_needing_notification_sent_between(from: from, to: to)
```